datafusion/functions/src/string/concat.rs

```diff
@@ -64,13 +66,18 @@ impl ScalarUDFImpl for ConcatFunc {
         &self.signature
     }
 
-    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
-        Ok(Utf8)
+    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
+        use DataType::*;
+        Ok(match &arg_types[0] {
+            LargeUtf8 => LargeUtf8,
+            _ => Utf8,
+        })
```

Comment:
currently Utf8View isn't declared as an expected return type, 
but the code does return Utf8View values.
should anything be failing when the runtime value type doesn't match the declared return type?
(if so, what would be a test case exercising such case?)

cc @comphead

Reply:
I've added test cases for Utf8View and LargeUtf8 return types. Have gone ahead and adjusted this.